File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        nums1.resize(m);
        for (int i = 0; i < n; i++) {
            nums1.push_back(nums2[i]);
        }
        int size = nums1.size();
        for (int i = 0; i < size - 1; i++) {
            for (int j = 0; j < size - i - 1; j++) {
                if (nums1[j] > nums1[j + 1]) {
                    swap(nums1[j], nums1[j + 1]);
                }
            }
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int s = m + n;
        int[] result = new int[s];

        int k = 0;
        int j = 0;

        for (int i = 0; i < s; i++) {

            if (k == m) {
                result[i] = nums2[j];
                j++;
            }

            else if (j == n) {
                result[i] = nums1[k];
                k++;
            }

            else if (nums1[k] <= nums2[j]) {
                result[i] = nums1[k];
                k++;
            }

            else {
                result[i] = nums2[j];
                j++;
            }
        }

        for (int i = 0; i < s; i++) {
            nums1[i] = result[i];
        }
    }
}